internal/csi/app/app.go

```diff
@@ -59,16 +60,28 @@ func NewCommand(ctx context.Context) *cobra.Command {
 				log.Info("propagating root CA bundle disabled")
 			}
 
+			annotations := map[string]string{}
+			for _, annotation := range opts.CertManager.CertificateRequestAnnotations {
+				if strings.Contains(annotation, "=") {
+					keyValue := strings.Split(annotation, "=")
+					if len(keyValue) != 2 {
+						return fmt.Errorf("certificaterequest annotation malformed on %s: expecting key value pair using '=' delimiter", annotation)
+					}
+					annotations[keyValue[0]] = keyValue[1]
```

Comment:
suggestion: Would it be worth adding a "reserved" prefix here? We use the `spiffe.csi.cert-manager.io/identity` annotation, for example, and it could be confusing if users are able to specify other annotations with a key like that.

Put another way, if I set a custom annotation `spiffe.csi.cert-manager.io/identity=myidentity`, which value is correct?

Maybe we can reject any custom annotation key which starts with `spiffe.csi.cert-manager.io`?